src/ppx_deriving.cppo.ml

```diff
@@ -234,6 +234,33 @@ let attr_warning expr =
   let structure = {pstr_desc = Pstr_eval (expr, []); pstr_loc = loc} in
   {txt = "ocaml.warning"; loc}, PStr [structure]
 
+let attr_nobuiltin ~deriver attrs =
+  attrs |> attr ~deriver "nobuiltin" |> Arg.get_flag ~deriver
+let rec remove_pervasive_lid = function
+  | Lident _ as lid -> lid
+  | Ldot (Lident "Pervasives", s) -> Lident s
+  | Ldot (lid, s) -> Ldot (remove_pervasive_lid lid, s)
+  | Lapply (lid, lid2) ->
+    Lapply (remove_pervasive_lid lid, remove_pervasive_lid lid2)
```

Comment:
I think this isn't right. It does too much. `Foo.Pervasives.x` shouldn't be reduced to `Foo.x` or at all.